lib/FusionInventory/Agent/Task/Inventory/Win32/License.pm

```diff
@@ -100,13 +90,32 @@ sub _getWmiSoftwareLicensingProducts {
             FULLNAME  => $object->{'Description'},
             NAME      => $object->{'Name'}
         };
-
-        if ($seenKey) {
-            # Update FULLNAME if seen ProductName in registry
-            $license->{FULLNAME} = encodeFromRegistry($seenKey->{'/ProductName'})
-                if $seenKey->{'/ProductName'};
-            $license->{TRIAL} = 1
-                if ($seenKey->{'/ProductNameBrand'} && $seenKey->{'/ProductNameBrand'} =~ /trial/i);
+        
+        if ($object->{'ID'} && $seenProducts->{lc($object->{'ID'})}) {
+            my $seenKey = $seenProducts->{lc($object->{'ID'})};
+            if ($seenKey->{'/ProductCode'}) {
+                my $ProductCodeUuid =  lc( $seenKey->{'/ProductCode'} =~ /([-\w]+)/ && $1 );
+                # Check Office registry is true
+                if ($seenProducts->{$ProductCodeUuid} && $seenProducts->{$ProductCodeUuid}->{'/DigitalProductID'}) {
+                    # Find Office registry licence
+                    my $passLicence = 0;
+                    for my $indexLicense (0 .. $#licences) {
```

Comment:
Using an array ref, you should better use a foreach here like: `foreach my $seen (@{$licenses}) {`
In the loop, you should set a `$found` hash ref in place of deleting an array element. This is always dangerous to delete and array element of the array we are looping on. Even if you do that and just leave the loop, this is just a wrong practice.
Keeping a found hash ref set, you can decide on l.121 to merge/replace the hash ref in place of adding it.